tests/test_fastapi_extension.py

```diff
@@ -335,3 +335,25 @@ async def t2(request: Request):
 
             response = client.get("/t2", headers={"foo": "5"})
             assert response.status_code == 200 if i < 6 else 429
+
+    @pytest.mark.parametrize(
+        "key_style, expected_key",
+        [
+            ("url", "LIMITER/mock//t1/1/1/minute"),
+            (
+                "endpoint",
+                "LIMITER/mock/tests.test_fastapi_extension.t1_func/1/1/minute",
+            ),
+        ],
+    )
+    def test_key_style(self, build_fastapi_app, key_style, expected_key):
+        app, limiter = build_fastapi_app(key_func=lambda: "mock", key_style=key_style)
+
+        @app.get("/t1")
```

Comment:
To match your explanation in the docs above, I feel like it would help to use an endpoint with a parameter in it, and test that 2 requests with different param values indeed lead to the behaviour you describe, depending on the value of `key_style`.

Reply:
good idea!